### PR TITLE
Start extracting a cycle path layer from OSM

### PR DESCRIPTION
**Pull request by dabreegster**
See https://github.com/acteng/atip/issues/299#issuecomment-1680872388 for notes/TODOs (easier to track there). Output at https://protomaps.github.io/PMTiles/?url=https%3A%2F%2Fatip.uk%2Flayers%2Fv1%2Fcycle_paths.pmtiles; a PR on the frontend coming soon.

This has many known limits, but it's a start. A good next step would be to start table-driven tests like https://github.com/a-b-street/osm2streets/blob/main/osm2streets/src/lanes/tests.rs, to have more confidence about weird edge cases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +117,8 @@ You can debug a PMTiles file using <https://protomaps.github.io/PMTiles>.
 
 There's a manual step required to generate `--wards`, `--census_output_areas`, and `--imd`. See the comment in the code.
 
+For `--cycle_paths`, you'll need about 20GB of RAM, until we switch to a streaming JSON parser.
```
Comment (dabreegster) [inline README.md:120]: https://pypi.org/project/ijson/ could help, but we need to set up poetry or venv for dependencies

File: layers/cycle_paths.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    cleanUpGeojson(
+        gjPath, getProps, filterFeatures=lambda f: getProps(f["properties"]) != None
+    )
+
+    convertGeoJsonToPmtiles(f"{tmp}/cycle_paths.geojson", "output/cycle_paths.pmtiles")
+
+
+# If this has some kind of cycle-path, returns a dictionary with:
+#
+# - kind = track | lane | shared_use_segregated | shared_use_unsegregated
```
Comment (dabreegster) [inline layers/cycle_paths.py:36]: This is a vastly simplified schema. I bounced around a bunch of other ideas and this is where I landed.

File: layers/cycle_paths.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            "osm_id": props["@id"],
+            "width": getSeparateWayWidth(props),
+            "direction": getSeparateWayDirection(props),
+        }
+    else:
+        return None
+
+
+def getSeparateWayWidth(props):
+    # TODO Handle suffixes / units
```
Comment (dabreegster) [inline layers/cycle_paths.py:78]: They're very rare, according to taginfo

File: layers/cycle_paths.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+def hasLane(props):
+    # TODO Handle bicycle:lanes
+    for suffix in ["", ":left", ":right", ":both"]:
+        if valueIndicatesLane(props.get("cycleway" + suffix)):
+            return True
+    return False
+
+
+# If there are two lanes, the result could capture one or both of the lanes
```
Comment (Pete-Y-CS) [inline layers/cycle_paths.py:108]: I suppose both here is not the width of both the lanes added together? I guess if left and right have different widths it's quite hard to get a good answer here

**Review by Pete-Y-CS — COMMENTED** [review]
Just one comment about the getLaneWidth stuff, otherwise very nice

**Review by Pete-Y-CS — APPROVED** [review]
I think given we don't know how it's gonna be used yet, the question around lane width can wait
